### PR TITLE
Implement macro that implements `TryBinaryOperator`,

### PR DESCRIPTION
**Pull request by ever0de**
1. In the binary operator of numeric , always calculate by changing `Rhs` based on the type of `Lhs`.
2. Only `try_multiply` is implemented for the `Interval` type, and the `Interval` type currently uses `std::ops::Mul` because there is no wrapping implementation for `checked_*` .
3. The `checked_*` functions only exist for integers, not floats. A float is converted to inf when it overflows. So we created a new `integer` module.

### CODE REVIEW

**Comment by ever0de** [comment]
Since the test code can also be replaced with a macro, we will work on it.

**Comment by ever0de** [comment]
And if the current dividing value of the div is 0, an Overflow error is raised.

**Comment by ChobobDev** [comment]
Thank you for the fantastic work!
Now we can add more data types as we desire with ease!!!

**Comment by ever0de** [comment]
rebase upstream

File: core/src/data/value/binary_op/integer/macros.rs [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+                    overflow_err($variant($primitive::MAX), I128(0), Divide)
+                );
+                assert_eq!(
+                    $primitive::MAX.try_divide(&U8(0)),
+                    overflow_err($variant($primitive::MAX), U8(0), Divide)
+                );
+            }
+
+            #[test]
+            fn mod_overlfow() {
```
Comment (panarch) [inline core/src/data/value/binary_op/integer/macros.rs:409]: tiny typo correction :) 
`mod_overflow`

**Review by panarch — APPROVED** [review]
This is... great work that will make us keep adding numeric data types easily.
Thanks a lot for the meaningful contribution.
Merging 👍 👍 👍

**Pull request merged**